src/Http/Auth/RequestAuthenticatorMiddleware.php

```diff
@@ -83,14 +83,15 @@ public function authenticate(ServerRequestInterface $request): void
     {
         // For requests to public routes, that provide extra information to Authenticated
         // Users. eg. show contributors
-        $this->checkToken($request);
-
-        if ($this->isCorsPreflightRequest($request) || $this->isPublicRoute($request)) {
-            return;
+        try {
+            $this->authenticateToken($request);
+        } catch (\Exception $exception) {
+            if ($this->isCorsPreflightRequest($request) || $this->isPublicRoute($request)) {
+                return;
```

Comment:
Probably best to also do `$this->token = null;` first like you did in `checkToken()`, since otherwise some parts of the code may think that the token was successfully authenticated on these public/CORS routes and incorrectly return the `contributors` property for example (depending on where the authentication failed).